src/amqp/publisher.ts

```diff
@@ -13,24 +13,17 @@ export class AMQPPublisher {
   }
 
   public async publish(exchange: string, routingKey: string, data: any): Promise<void> {
-    let promise: PromiseLike<amqp.Channel>;
-    if (this.channel) {
-      promise = Promise.resolve(this.channel);
-    } else {
-      promise = this.connection.createChannel();
+    const channel = await this.getOrCreateChannel();
+    await channel.assertExchange(exchange, 'topic', { durable: false, autoDelete: false });
+    await channel.publish(exchange, routingKey, Buffer.from(JSON.stringify(data)));
```

Comment:
`try/catch` is not needed. Errors here will be thrown to higher stack level `await/catch`'s or `try/catch`blocks. In this specific case, consumers in application code calling the pubsub `publish` would be responsible for deciding how to handle this lower level error that has bubbled up.

This will be unhandled if that call was not `await`'d or `catch`'d

Reply:
This is a good summary of how that works: https://javascript.info/async-await#error-handling